boards/riscv/renode_riscv32/doc/index.rst

```diff
@@ -0,0 +1,55 @@
+.. _renode-riscv32:
+
+Renode RISCV32
```

Comment:
Hi! While Renode normally targets real-life hardware (i.e. same software is run on hw and in sim), it also allows its users to create virtual platforms, or to model hardware which is yet to be released.

I would suggest you to change the name of this virtual platform to something less generic sounding, so that it's less confusing for the user. I'm worried that it might be confusing for the user to see e.g. commands such as `west build -b hifive1 -t run_renode` and then see that there's a separate platform called `renode_riscv32` and cause them to wonder - how are these two things related?

If you have a name for this hw (any identification string, really), I'd suggest you to change the name of this Zephyr target to something like `<name>_virtual`. I see in the YAML file that you're referencing `m2gl025_miv` there. So maybe something like `m2gl025_miv_virtual`, and adding a description saying that this platform is derived from the `m2gl025_miv` Zephyr target would be a good idea?

Reply:
My original intention was to create a board for [riscv_virt](https://github.com/renode/renode/blob/master/platforms/cpus/riscv_virt.repl) rather than `m2gl025_miv`, wouldn't naming it as `m2gl025_miv_virtual` cause confusion to the user of `m2gl025_miv`?

Reply:
Ah, okay then, because `boards/riscv/renode_riscv32/renode_riscv32.yaml` looked really familiar to its counterpart for the `m2gl025_miv` board (e.g. `vendor` set to Microchip) to me. Maybe renaming the board to `riscv_virtual`, or `riscv_custom` would be fine with you?

Reply:
> Maybe renaming the board to `riscv_virtual`, or `riscv_custom` would be fine with you?

`riscv_virtual` sounds good to me, how about the `soc`? Is the current name ok for you?

Reply:
I've changed the `soc` from `renode` to `renode_virt`